holidays/countries/hongkong.py

```diff
@@ -38,173 +41,379 @@ class HongKong(
+                self._add_holiday(
+                    "The day following Ching Ming Festival", dt_qingming + td(days=+1)
+                )
+            else:
+                # 清明節.
+                self._add_qingming_festival("Ching Ming Festival")
 
-        # Chung Yeung Festival
-        self._add_double_ninth_festival("Chung Yeung Festival")
+        # Good Friday.
+        # 耶穌受難節.
+        self._add_good_friday("Good Friday")
 
-        # National Day
-        if self._year >= 1997:
-            self._add_holiday_oct_1("National Day")
+        # The day following Good Friday.
+        # 耶穌受難節翌日.
+        self._add_holy_saturday("The day following Good Friday")
 
-            if self._year <= 1998:
-                self._add_holiday_oct_2("National Day")
+        # Easter Monday.
+        if self._year >= 1968 and dt_qingming == self._easter_sunday:
+            # 復活節星期一翌日.
+            self._add_holiday_2_days_past_easter("The day following Easter Monday")
+        else:
+            # 復活節星期一.
+            self._add_easter_monday("Easter Monday")
 
-        # Christmas Day
+        # The Birthday of the Buddha.
+        if self._year >= 1999:
+            dt_birthday_of_buddha = self._chinese_calendar.buddha_birthday_date(self._year)[0]
+            if self._is_sunday(dt_birthday_of_buddha):
+                # 佛誕翌日.
+                self._add_holiday(
+                    "The day following the Birthday of the Buddha",
+                    dt_birthday_of_buddha + td(days=+1),
+                )
+            else:
+                # 佛誕.
+                self._add_chinese_birthday_of_buddha("The Birthday of the Buddha")
+
+        # Labour Day.
+        if self._year >= 1999:
+            if self._is_sunday(MAY, 1):
+                # 勞動節翌日.
+                self._add_labor_day_two("The day following Labour Day")
+            else:
+                # 勞動節.
+                self._add_labor_day("Labour Day")
+
+        # Tuen Ng Festival.
+        if self._year >= 1968:
+            dt_dragon_boat = self._chinese_calendar.dragon_boat_date(self._year)[0]
+            if self._is_sunday(dt_dragon_boat):
+                # 端午節翌日.
+                self._add_holiday(
+                    "The day following Tuen Ng Festival", dt_dragon_boat + td(days=+1)
+                )
+            else:
+                # 端午節.
+                self._add_dragon_boat_festival("Tuen Ng Festival")
+
+        # Hong Kong Special Administrative Region Establishment Day.
+        if self._year <= 1982 or self._year >= 1997:
+            if self._is_sunday(JUL, 1):
+                # 香港特別行政區成立紀念日翌日.
+                self._add_holiday_jul_2(
+                    "The day following Hong Kong Special Administrative Region Establishment Day"
+                )
+            else:
+                # 香港特別行政區成立紀念日.
+                self._add_holiday_jul_1(
+                    "Hong Kong Special Administrative Region Establishment Day"
+                )
```

Comment:
Yes, it seems I was wrong. According to [Wiki](https://zh.wikipedia.org/wiki/%E9%A6%99%E6%B8%AF%E7%AF%80%E6%97%A5%E8%88%87%E5%85%AC%E7%9C%BE%E5%81%87%E6%9C%9F#%E5%85%AC%E7%9C%BE%E5%81%87%E6%9C%9F_2) (Google translated):

> Since 1912, the government has declared the first Sunday in July (usually July 1, if it is a Sunday, Monday, July 2) as a public holiday.  By 1923, it would be This arrangement sets a precedent.
> ...
> Since 1983 ... Two additional holidays will be added to replace the public holidays on the first Sunday in July and the first Monday in August.

there was a holiday on the first Sunday in July, but since I didn't find any information about it, it's reasonable not to include it.